### PR TITLE
Alicia news task4

### PR DESCRIPTION
**Pull request by AliciaSarah92**

### CODE REVIEW

**Comment by saimakhanom** [comment]
It looks like you may be completing tasks 4 and 5 on the same branch - be sure to always check which branch you are working on before getting started

File: __tests__/app.test.js [hunk 1]
```diff
@@ -42,4 +43,23 @@ describe('GET /api/articles', () => {
...
+            .get('/api/articles')
+            .then(({ body }) => {
+                expect(body.articles.length).toBeGreaterThan(0);
+                expect(body.articles[0]).toEqual(
+                    expect.objectContaining({
+                        article_id: expect.any(Number),
+                        title: expect.any(String),
+                        topic: expect.any(String),
+                        author: expect.any(String),
+                        body: expect.any(String),
```
Comment (saimakhanom) [inline __tests__/app.test.js]: the /api/articles endpoint should return articles without the body property, and with an additional property called comment_count

File: __tests__/app.test.js [hunk 2]
```diff
@@ -55,18 +55,29 @@ describe('GET /api/articles', () => {
...
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String)
+                })
+            );
+        })
+    })
+    test('should return a 404 error if incorrect article_id does not exist', () => {
```
Comment (saimakhanom) [inline __tests__/app.test.js:78]: we also need to test if the article id provided is invalid, how would the server respond? e.g. /api/articles/bananas

File: controllers/articles.controller.js [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+};
+
+exports.getArticle = (req, res) => {
+    const { article_id } = req.params;
+    if (isNaN(article_id)) {
+        res.status(400).send({ msg: 'bad request' });
+    }
+    selectArticle(article_id).then(article => {
+        if (!article) {
+            res.status(404).send({ msg: 'record does not exist' });
```
Comment (saimakhanom) [inline controllers/articles.controller.js:16]: we should reject the promise at this point and allow the error handling middleware to send the response to the user

File: __tests__/app.test.js [hunk 4]
```diff
@@ -50,34 +52,34 @@ describe('GET /api/articles', () => {
...
         return request(app)
-            .get('/api/articles')
-            .then(({ body }) => {
-                expect(body.articles.length).toBeGreaterThan(0);
+            .get('/api/articles/1')
+            .expect(200)
+            .then(response => {
+                expect(response.body).toEqual(
+                    expect.objectContaining({
+                        article_id: expect.any(Number),
```
Comment (MRombi) [inline __tests__/app.test.js:62]: This should be 1 not any number 🙂

**Review by MRombi — APPROVED** [review]
Good job implementing the feedback! Please make the change I mentioned and merge. No need to resubmit 🙂

**Pull request merged**